app/site/_includes/tag.liquid

```diff
@@ -0,0 +1,46 @@
+<script>
+    document.addEventListener('DOMContentLoaded', function() {
+      // Function to set the background color based on project type
+      
+        function setTagColor(projType) {
+            let color;
+            switch (projType) {
+            case 'Tools':
+                color = 'green';
+                break;
+            case 'APIs':
+                color = 'red';
+                break;
+            case 'Apps':
+                color = 'blue';
+                break;
+            case 'Data':
+                color = 'orange';
+                break;
+            case 'Docs':
+                color = 'mint';
+                break;
+            case 'Libraries':
+                color = 'magenta';
+                break;
+            case 'Standards':
+                color = 'gold';
+                break;
+            default:
+                color = '#1a4480';
+            }
+
+            // Set the CSS variable on the element with the class 'main-tag'
+            const element = Array.from(document.querySelectorAll('span')).find(span => span.textContent.trim() === projType);
+
+            if (element) {
+            element.style.setProperty('--tag-color', color);
+            } else {
+            console.error('Element with class "project type tag" not found');
+            }
+        }
+
+        var projType = '{{ project.project_type }}';
+        setTagColor(projType);
```

Comment:
Would it be easier to add all of this to CSS if we already need a list up front? So maybe add a filter to slugify the tag names in case we add any with spaces later (i.e. "Test topic" becomes something like "test-topic") and then just add a line for `.test-topic-tag`. There might be a built-in filter

Reply:
ended up applying this method! 🎉